src/utils/cspr-network.ts

```diff
@@ -0,0 +1,208 @@
+import { RpcClient } from '../rpc';
+import {
+  Args,
+  CLValue,
+  CLValueUInt512,
+  ContractCallBuilder,
+  NativeDelegateBuilder,
+  NativeRedelegateBuilder,
+  NativeTransferBuilder,
+  NativeUndelegateBuilder,
+  PublicKey,
+  Transaction,
+  TransactionHash
+} from '../types';
+import { BigNumber } from '@ethersproject/bignumber';
+
+export class CasperNetwork {
+  private rpcClient: RpcClient;
+  private apiVersion: number;
+
+  constructor(rpcClient: RpcClient, apiVersion: number) {
+    this.rpcClient = rpcClient;
+    this.apiVersion = apiVersion;
+  }
+
+  public static async create(
+    rpcClient: RpcClient,
+    apiVersion?: number
+  ): Promise<CasperNetwork> {
+    if (!apiVersion) {
+      const status = await rpcClient.getStatus();
+
+      apiVersion = status.apiVersion.startsWith('2.') ? 2 : 1;
+    }
+
+    return new CasperNetwork(rpcClient, apiVersion);
+  }
+
+  public createDelegateTransaction(
+    delegatorPublicKey: PublicKey,
+    validatorPublicKey: PublicKey,
+    networkName: string,
+    amountMotes: string | BigNumber,
+    deployCost: number,
+    ttl: number,
+    contractHash?: string
+  ) {
+    if (this.apiVersion === 2) {
+      new NativeDelegateBuilder()
+        .validator(validatorPublicKey)
+        .from(delegatorPublicKey)
+        .amount(amountMotes)
+        .chainName(networkName)
+        .payment(deployCost)
+        .ttl(ttl)
+        .build();
+    }
+
+    if (contractHash) {
+      return new ContractCallBuilder()
+        .from(delegatorPublicKey)
+        .byHash(contractHash)
+        .entryPoint('delegate')
+        .chainName(networkName)
+        .runtimeArgs(
+          Args.fromMap({
+            validator: CLValue.newCLPublicKey(validatorPublicKey),
+            delegator: CLValue.newCLPublicKey(delegatorPublicKey),
+            amount: CLValueUInt512.newCLUInt512(amountMotes)
+          })
+        )
+        .ttl(ttl)
+        .buildFor1_5();
+    }
+
+    return new Error('Need to provide contract hash');
```

Comment:
Let's change to `Auction contract hash is required when creating a transaction on Casper Network 1.5.x`